static/js/user_group_edit.js

```diff
@@ -112,4 +118,59 @@ export function initialize() {
             open_group_edit_panel_for_row(this);
         }
     });
+
+    $("#manage_groups_container").on("click", "#open_group_info_modal", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        const user_group_id = get_user_group_id(e.target);
+        const user_group = user_groups.get_user_group_from_id(user_group_id);
+        const template_data = {
+            group_name: user_group.name,
+            group_description: user_group.description,
+        };
+        const change_user_group_info_modal = render_change_user_group_info_modal(template_data);
+        dialog_widget.launch({
+            html_heading: $t_html(
+                {defaultMessage: "Edit {group_name}"},
+                {group_name: user_group.name},
+            ),
+            html_body: change_user_group_info_modal,
+            id: "change_group_info_modal",
+            on_click: save_group_info,
+            post_render: () => {
+                $("#change_group_info_modal .dialog_submit_button")
+                    .addClass("save-button")
+                    .attr("data-group-id", user_group_id);
+            },
+        });
+    });
+
+    function save_group_info(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        const group = get_user_group_for_target(e.currentTarget);
+
+        const url = `/json/user_groups/${group.id}`;
+        const data = {};
+        const new_name = $("#change_user_group_name").val().trim();
+        const new_description = $("#change_user_group_description").val().trim();
+
+        if (new_name === group.name && new_description === group.description) {
+            return;
+        }
+        if (new_name !== group.name) {
+            data.name = new_name;
+        } else {
+            data.name = group.name;
```

Comment:
Should we instead be not passing the `name` parameter when it's unchanged?

Reply:
We need to pass name and description even when they are not edited because they are required parameter for the api, and if not passed in the request the default is empty string, so if only `description` is edited and we send only `description` in the api then the request updates the user group name with empty string. Attaching a snip of the view that handles the user_group patch request.

![image](https://user-images.githubusercontent.com/63504956/188221862-c2f59ce8-4af8-44bd-bf86-4daa5785f7dd.png)

Reply:
Yikes, that sounds like a bug. Can you start an "#api design" thread about this? I think we likely want to change both parameters to optional and not request a change if the parameter wasn't passed, but it's worth the visibility of posting there.

(And then we can clean up this logic as part of the PR to fix that, I think).